frame/assets/src/impl_fungibles.rs

```diff
@@ -77,6 +77,13 @@ impl<T: Config<I>, I: 'static> fungibles::InspectMetadata<<T as SystemConfig>::A
 	fn decimals(asset: &Self::AssetId) -> u8 {
 		Metadata::<T, I>::get(asset).decimals
 	}
+
+	/// Return all assets id by name.
+	fn asset_ids(asset_name: Vec<u8>) -> Vec<Self::AssetId> {
+		Metadata::<T, I>::iter()
```

Comment:
Just my 2 cents: Honestly not a fan of this. `StorageMap::iter()` has unbounded weight, which makes it a footgun in on-chain development.
If you really desperately need this, I would add a separate trait with the appropriate warnings of "Unbouded execution time, DO NOT use without knowing what you're doing" etc.
(Note that none of the other traits in this file use `::iter()`.)
If you want to add this for an RPC, then I can see the motivation, but maybe would be better to add it with the RPC?

Reply:
Also not a fan of accessing assets by name. The name can be changed any time, so it's hard to make this dependable.
I would recommend against building on-chain functionality that looks up assets by name.

Reply:
> Honestly not a fan of this. `StorageMap::iter()` has unbounded weight, which makes it a footgun in on-chain development.

That is a good argument! I have overseen this!